src/sentry/incidents/endpoints/organization_alert_rule_index.py

```diff
@@ -37,14 +45,47 @@ def get(self, request, organization):
                 "id", flat=True
             )
 
+        teams = set(request.GET.getlist("team", []))
+        team_filter_query = None
+        if teams:
+            request_teams = request.access.teams
+            for team in request_teams:
+                if not request.access.has_team_access(team):
+                    return Response(
+                        f"Error: You do not have permission to access {team.name}",
+                        status=status.HTTP_400_BAD_REQUEST,
+                    )
+
+            unassigned = None
+            if "unassigned" in teams:
+                teams.remove("unassigned")
+                unassigned = Q(owner_id=None)
+
+            if "myteams" in teams:
+                teams.remove("myteams")
+                teams.update([t.id for t in request_teams])
+
+            for team_id in teams:  # Verify each Team id is numeric
```

Comment:
What I want to validate is that any IDs directly passed via query param (as opposed to teams we fetch with "myteams") need to be checked with `request.access.has_team_access`

Reply:
🤦